sdk/python/feast/infra/gcp.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from typing import List, Optional
+
+from feast import FeatureTable
+from feast.infra.provider import Provider
+from feast.repo_config import DatastoreOnlineStoreConfig
+
+
+def _delete_all_values(client, key) -> None:
+    """
+    Delete all keys under the key path in firestore.
+    """
+    while True:
+        query = client.query(kind="Value", ancestor=key)
+        entities = list(query.fetch(limit=1000))
+        if not entities:
+            return
+
+        for entity in entities:
+            print("Deleting: {}".format(entity))
+            client.delete(entity.key)
+
+
+class Gcp(Provider):
+    _project_id: Optional[str]
+
+    def __init__(self, config: Optional[DatastoreOnlineStoreConfig]):
+        if config:
+            self._project_id = config.project_id
+        else:
+            self._project_id = None
+
+    def _initialize_client(self):
+        from google.cloud import datastore
+
+        if self._project_id is not None:
+            return datastore.Client(self.project_id)
+        else:
+            return datastore.Client()
+
+    def update_infra(
```

Comment:
Once we've swept through PR comments, can we get Google style docstrings for all public methods?

Reply:
This one has a comment on the base abstract interface, so i didn't duplicate it here (it is just a gcp specific implementation)